src/middle/Stan_math_signatures.ml

```diff
@@ -523,6 +523,14 @@ let () =
     ("block", ReturnType UMatrix, [UMatrix; UInt; UInt; UInt; UInt]) ;
   add_unqualified ("categorical_rng", ReturnType UInt, [UVector]) ;
   add_unqualified ("categorical_logit_rng", ReturnType UInt, [UVector]) ;
+  add_unqualified
+    ( "categorical_logit_glm_lpmf"
+    , ReturnType UReal
+    , [bare_array_type (UInt, 1); UMatrix; UVector; UMatrix] ) ;
```

Comment:
I was trying to add signatures for both int and std::vector<int>. This add just the latter right? Would DVInt be the right choice here? Or just make a duplicate?

Reply:
Supported signatures should be: 
```
real categorical_logit_glm_lpmf(int, matrix, vector, matrix)
real categorical_logit_glm_lpmf(int, row_vector, vector, matrix)
real categorical_logit_glm_lpmf(int[], matrix, vector, matrix)
real categorical_logit_glm_lpmf(int[], row_vector, vector, matrix)
```

Reply:
How does it take both an it or an array of ints with the same last 3 arguments?

Reply:
The signature with `int` and `row_vector` calculates stuff for a single instance. The signature with `int[]` and `matrix` does same for multiple instances. Other two signatures can broadcast arfuments - use same values of either first or second argument for all instances. 

More info in discussion that started here: https://github.com/stan-dev/math/pull/1242#issuecomment-495637043

Reply:
That discussion says that the 2nd parameter should be a matrix only if y is an array,
> y can be a scalar OR array of size n,
> x can be a vector of size J OR n x J matrix (only if y is an array)